sound/soc/bcm/pisound.c

```diff
@@ -581,13 +585,25 @@ static int spi_read_info(void)
 				return -EINVAL;
 
 			snprintf(
-				g_version,
-				sizeof(g_version),
+				g_fw_version,
```

Comment:
Thank you for thoroughly looking into the changes. Actually it will always succeed - the buffer is 6 bytes long, the fmt string is "%x.%02x" and another similar one is "%x.%x", that means the longest possible string where the arguments are a single byte each would be for example "ff.ff" (and implicit \0 byte). That fits exactly into the 6 bytes available. It would become a problem if the arguments would be larger than 'char' however, but this location is not likely to change element size.

Reply:
It's still safer to always set the last byte of the buffer to NUL just in case.